code/modules/mob/living/carbon/human/human_powers.dm

```diff
@@ -1122,3 +1122,40 @@ mob/living/carbon/human/proc/change_monitor()
 		to_chat(src, span("notice", "You sense " + jointext(feedback, " ") + " towards the [dir2text(text2num(d))]."))
 	if(!length(dirs))
 		to_chat(src, span("notice", "You detect no psionic signatures but your own."))
+
+// flick tongue out to read gasses
+/mob/living/carbon/human/proc/tongue_flick()
+	set name = "Tongue-flick"
+	set desc = "Flick out your tongue to sense the gas in the room."
+	set category = "IC"
+
+	if(stat == DEAD)
+		return
+
+	if(last_special > world.time)
+		return
+
+	if(head && (head.body_parts_covered & FACE))
+		to_chat(src, span("notice", "You can't flick your tongue out with something covering your face."))
+		return
+	else
+		say("!flicks their tongue out.")
+
+	var/datum/gas_mixture/mixture = src.loc.return_air()
+	var/total_moles = mixture.total_moles
+
+	var/list/airInfo = list()
+	if(total_moles>0)
+		for(var/mix in mixture.gas)
+			airInfo += span("notice", "[gas_data.name[mix]]: [round((mixture.gas[mix] / total_moles) * 100)]%")
```

Comment:
I am also not too sure about the exact gas information down to the moles.
Imho it would be preferable to just give general information about the presence of gasses in the mix.

Reply:
Borrowed some code from my medical health analyzer nerf, it now displays rough estimates at various gas levels.
if(0)
   composition = "Non-existant"
if(0 to 5)
   composition = "Trace-amounts"
if(5 to 15)
   composition = "Low-scent"
if(15 to 60)
   composition = "Present"
if(60 to 80) //nitrogen is usually at 78%
   composition = "High-volume"
if(80 to INFINITY)
   composition = "Overwhelming"